service/validator.go

```diff
@@ -113,9 +129,63 @@ func (sv ValidatorConfig) ValidateRequestAndReturnToken(r *http.Request) (TokenW
 		token: token,
 	}
 
+	err = sv.validateAudiences(*tokenWithClaims, sv.audience.All())
+	if err != nil {
+		return nil, err
+	}
+
 	return tokenWithClaims, nil
 }
 
+// ValidateAudiences ...
+func (sv ValidatorConfig) validateAudiences(tokenWithClaims tokenWithClaims, audiences []string) error {
+	payload, err := tokenWithClaims.Payload()
+	if err != nil {
+		return err
+	}
+
+	var audiencesInToken []string
+
+	switch v := payload["aud"].(type) {
+	default:
+		fmt.Printf("unexpected type %T", v)
+	case string:
+		audience, ok := payload["aud"].(string)
+		if !ok {
+			panic("type assertion failed")
+		}
+		audiencesInToken = []string{audience}
+	case []interface{}:
+		audienceInterface, ok := payload["aud"].([]interface{})
+		if !ok {
```

Comment:
At this point you don't need the safe type assertion

Reply:
are you sure? I mean **errcheck** fails if you don't check the result

Reply:
See my other response